src/support/cleanse.cpp

```diff
@@ -5,9 +5,36 @@
 
 #include "cleanse.h"
 
-#include <openssl/crypto.h>
+#include <string.h>
 
+/* Use asm directives to protect OPENSSL_cleanse.
+ *
+ * Compilers have a bad habit of removing "superfluous" memset calls that
+ * are trying to zero memory. For example, when memset()ing a buffer and
+ * then free()ing it, the compiler might decide that the memset is
+ * unobservable and thus can be removed.
+ *
+ * Previously we tried to stop this by a) implementing memset in assembly
+ * on x86 and b) putting the function in its own file for other platforms.
+ *
+ * This change removes those tricks in favour of using asm directives to
+ * scare the compiler away. As best as our compiler folks can tell, this is
+ * sufficient and will continue to be so.
+ *
+ * Adam Langley <agl@google.com>
+ * Commit: ad1907fe73334d6c696c8539646c21b11178f20f
+ * BoringSSL (LICENSE: ISC)
+ */
 void memory_cleanse(void *ptr, size_t len)
 {
-    OPENSSL_cleanse(ptr, len);
+    memset(ptr, 0, len);
+
+    /* As best as we can tell, this is sufficient to break any optimisations that
+       might try to eliminate "superfluous" memsets. If there's an easy way to
+       detect memset_s, it would be better to use that. */
+#if defined(OPENSSL_WINDOWS)
```

Comment:
I'm not sure OpenSSL's headers expose such a macro, and even if they do, this risks breaking things if OpenSSL gets removed entirely at some point. I believe you just want `_MSC_VER` here to detect MSVC (as the `else` branch should work fine in MinGW).